packages/react/src/exchanges/hooks/useExchangeFilters.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  createExchangeFilter as createExchangeFilterAction,
+  getExchangeFilterError,
+  getExchangeFilters,
+  isExchangeFilterLoading,
+  resetExchangeFilters,
+  type StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction.js';
+import type {
+  Config,
+  ExchangeFilterItem,
+  OrderItem,
+  PostExchangeFilterData,
+} from '@farfetch/blackout-client';
+
+type PostExchangeFilterDataHook = PostExchangeFilterData & {
+  exchangeFilterItems: Array<ExchangeFilterItemWithOptionalUuid>;
+};
+
+type ExchangeFilterItemWithOptionalUuid = Omit<
+  ExchangeFilterItem,
+  'orderItemUuid'
+> & {
+  orderItemUuid?: OrderItem['shippingOrderLineId'];
+};
+
+/**
+ * Provides facilities to create an exchange filter.
+ */
+function useExchangeFilters(orderItemUuid?: OrderItem['shippingOrderLineId']) {
+  const orderItemUuidHookParameter = orderItemUuid;
+  const isCreating = useSelector((state: StoreState) =>
```

Comment:
Since the selector that is being used here is called `isExchangeFilterLoading`, I would change this name to `isLoading`. What do you think? If you agree, do not forget to add this to the description of the commit as it will be a breaking change.

Reply:
Fixed and added to the description.